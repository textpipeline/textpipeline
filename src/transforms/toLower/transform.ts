import { Transform } from '../types';

const transform: Transform = {
  name: 'To Lowercase',
  slug: 'to-lower',
  inputTypeName: 'Text',
  outputTypeName: 'Lowercased Text',
  openSourceProject: undefined,
  defaultOutput: '',
  execute: async input => input.toLowerCase(),
};

export default transform;
