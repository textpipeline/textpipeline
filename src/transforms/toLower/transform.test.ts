import toLower from './transform';

describe('toLower', () => {
  it('has the correct attributes', () => {
    expect(toLower).toMatchSnapshot();
  });

  describe('execute', () => {
    it('converts empty to empty', () => expect(toLower.execute('')).resolves.toEqual(''));

    it('converts already lower cased text to itself', () =>
      expect(toLower.execute('foo bar')).resolves.toEqual('foo bar'));

    it('converts mixed case to lower case', () => expect(toLower.execute('Foo baR')).resolves.toEqual('foo bar'));

    it('converts upper case to lower case', () => expect(toLower.execute('FOO BAR')).resolves.toEqual('foo bar'));

    it('converts text with non-alpha values correctly', () =>
      expect(toLower.execute('FOO bar 123 $%^&')).resolves.toEqual('foo bar 123 $%^&'));
  });
});
