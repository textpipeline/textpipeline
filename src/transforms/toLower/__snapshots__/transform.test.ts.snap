// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toLower has the correct attributes 1`] = `
Object {
  "defaultOutput": "",
  "execute": [Function],
  "inputTypeName": "Text",
  "name": "To Lowercase",
  "openSourceProject": undefined,
  "outputTypeName": "Lowercased Text",
  "slug": "to-lower",
}
`;
